contracts/axelar-interchain-token-service/src/contract.rs

```diff
@@ -0,0 +1,100 @@
+use axelar_soroban_std::ensure;
+use soroban_sdk::{contract, contractimpl, Address, Env, String,};
+
+use crate::error::InterchainTokenServiceError;
+use crate::event;
+use crate::storage_types::{DataKey, TrustedItsAddress};
+
+#[contract]
+pub struct InterchainTokenService;
+
+#[contractimpl]
+impl InterchainTokenService {
+    pub fn initialize(
+        env: Env,
+        owner: Address
+    ) -> Result<(), InterchainTokenServiceError> {
+        ensure!(
+            env.storage()
+                .instance()
+                .get(&DataKey::Initialized)
+                .unwrap_or(true),
+            InterchainTokenServiceError::AlreadyInitialized
+        );
+
+        env.storage().instance().set(&DataKey::Initialized, &true);
+
+        env.storage().instance().set(&DataKey::Owner, &owner);
+
+        Ok(())
+    }
+
+    pub fn owner(env: &Env) -> Address {
+        env.storage().instance().get(&DataKey::Owner).unwrap()
+    }
+
+    pub fn transfer_ownership(env: Env, new_owner: Address) {
+        let owner = Self::owner(&env);
+        owner.require_auth();
+
+        env.storage().instance().set(&DataKey::Owner, &new_owner);
+
+        event::transfer_ownership(&env, owner, new_owner);
+    }
+
+    pub fn is_trusted_address(env: &Env, chain: String, address: String) -> bool {
```

Comment:
```suggestion
    pub fn trusted_address(env: &Env, chain: String) -> String {
```
Retrieve the trusted address instead